app/models/host/managed.rb

```diff
@@ -606,6 +606,7 @@ def puppetrun!
     ProxyAPI::Puppet.new({:url => puppet_proxy.url}).run fqdn
   rescue => e
     errors.add(:base, _("failed to execute puppetrun: %s") % e)
+    logger.warn "unable to execute puppet run: " % e
```

Comment:
logging e.backtrace to debug is also common, and would be useful here, e.g. https://github.com/theforeman/foreman/blob/develop/app/controllers/hosts_controller.rb#L673-L674